biogeochem/EDPhysiologyMod.F90

```diff
@@ -595,81 +645,99 @@ subroutine phenology_leafonoff(currentSite)
+                   currentCohort%status_coh = 2    ! Leaves are on, so change status to 
+                                                   ! stop flow of carbon out of bstore. 
+                   
+                   if(store_c>nearzero) then
+                      store_c_transfer_frac = &
+                            min(EDPftvarcon_inst%phenflush_fraction(ipft)*currentCohort%laimemory, store_c)/store_c
                    else
-                      ! we can only put on as much carbon as there is in the store...
-                      ! nb. Putting all of bstore into leaves is C-starvation suicidal. 
-                      ! The tendency for this could be parameterized
-                      currentCohort%bl = currentCohort%bstore * store_output
-                   endif
-
+                      store_c_transfer_frac = 0.0_r8
+                   end if
 
-                   if ( debug ) write(fates_log(),*) 'EDPhysMod 1 ',currentCohort%bstore
-
-                   currentCohort%bstore = currentCohort%bstore - currentCohort%bl  ! Drain store
-
-                   if ( debug ) write(fates_log(),*) 'EDPhysMod 2 ',currentCohort%bstore
+                   ! This call will request that storage carbon will be transferred to 
+                   ! leaf tissues. It is specified as a fraction of the available storage
+                   call PRTPhenologyFlush(currentCohort%prt, ipft, leaf_organ, store_c_transfer_frac)
 
                    currentCohort%laimemory = 0.0_r8
 
                 endif !pft phenology
              endif ! growing season 
 
              !COLD LEAF OFF
-!             currentCohort%leaf_litter = 0.0_r8 !zero leaf litter for today. 
              if (currentSite%status == 1)then !past leaf drop day? Leaves still on tree?  
                 if (currentCohort%status_coh == 2)then ! leaves have not dropped
-                   currentCohort%status_coh      = 1                  
-                   !remember what the lai was this year to put the same amount back on in the spring... 
-                   currentCohort%laimemory   = currentCohort%bl  
+                   
+                   ! This sets the cohort to the "leaves off" flag
+                   currentCohort%status_coh  = 1
+
+                   ! Remember what the lai was (leaf mass actually) was for next year
+                   ! the same amount back on in the spring...
+
+                   currentCohort%laimemory   = leaf_c
 
-                   ! add lost carbon to litter
-                   currentCohort%leaf_litter = currentCohort%bl 
-                   currentCohort%bl          = 0.0_r8   
+                   ! Drop Leaves (this routine will update the leaf state variables,
+                   ! for carbon and any other species that are prognostic. It will
+                   ! also track the turnover masses that will be sent to litter later on)
+
+                   call PRTDeciduousTurnover(currentCohort%prt,ipft, &
```

Comment:
at some point we should make this so that leaves can come on and off gradually, but that's for later...

Reply:
that would definitely be a nice feature